runtime-modules/council/src/lib.rs

```diff
@@ -706,6 +876,53 @@ impl<T: Trait> Mutations<T> {
     fn set_candidacy_note(membership_id: &T::MembershipId, note_hash: &T::Hash) {
         Candidates::<T>::mutate(membership_id, |value| value.note_hash = Some(*note_hash));
     }
+
+    /// Removes member's candidacy record.
+    fn clear_candidate(membership_id: &T::MembershipId) {
+        // clear candidate record
+        Candidates::<T>::remove(membership_id);
```

Comment:
It feels a bit risky to drop the candidates but leave the lock removal until later in the code path in `elect_new_council`. Perhaps its a bit tidier to move lock removal in here, and then that new method just applies the council lock?

Reply:
Yes, that's probably better. Done.